Lib/logging/__init__.py

```diff
@@ -1543,9 +1558,17 @@ def isEnabledFor(self, level):
         """
         Is this logger enabled for level 'level'?
         """
-        if self.manager.disable >= level:
-            return False
-        return level >= self.getEffectiveLevel()
+        try:
+            return self._cache[level]
+        except KeyError:
+            _acquireLock()
+            if self.manager.disable >= level:
```

Comment:
> I don't see a functional difference. Is there?

The `manager.disable` check is [duplicated] there because it would potentially allow skipping the hierarchy walk of `getEffectiveLevel()`.

> The alternative is to give up some performance and move caching to `getEffectiveLevel()`

Although it slows down `isEnabledFor()` because of still needing the `manager.disable` check, I think it would be better to cache `getEffectiveLevel()`. It's easier to reason about. The `manager.disable` level shouldn't be conflated with the effective levels of individual loggers, IMO.